src/Forms/__tests__/TextInput.test.tsx

```diff
@@ -111,6 +111,124 @@ describe('Text input', function () {
       strictEqual(input.hasAttribute('aria-invalid'), false);
     });
   });
+  context('when hideError and errorMessage prop are present', function () {
+    beforeEach(function () {
+      changeSpy = spy();
+      ({ queryByText, getByText } = render(
+        <TextInput
+          id="semester"
+          name="semester"
+          value="Spring"
+          label="semester"
+          errorMessage="Error: Please enter a valid ID"
+          hideError
+          onChange={changeSpy}
+        />
+      ));
+    });
+    it('renders', function () {
+      const inputElement = document.getElementById('semester') as HTMLInputElement;
+      strictEqual(!!inputElement, true);
+    });
+    it('calls the change handler when changed', function () {
+      fireEvent.change(document.getElementById('semester'), {
+        target: {
+          value: 'Fall',
+        },
+      });
+      strictEqual(changeSpy.callCount, 1);
+    });
+    it('renders the correct default value', function () {
+      const inputField = document.getElementById('semester') as HTMLInputElement;
+      const defaultValue = inputField.value;
+      strictEqual(defaultValue, 'Spring');
+    });
+    it('does not render the error message', function () {
+      const errorField = queryByText('error');
+      strictEqual(errorField, null);
+    });
```

Comment:
Is this the behavior we want that when there is an error message we don't show it? This seems like it may be an issue to hide the error when there is one. I thought the purpose of the `hideError` prop was to  adjust the formatting when there is no `errorMessage` present so that there isn't an empty space under the filter fields.

Reply:
We did not discuss that, therefore I (logic of not native english speaker) decide to hide the error message when hideError is set to true. Otherwise "hideError" will be misleading if we hide a non existing error.

Reply:
J Seitz confirmed that we should be able to hide the error when `errorMessage` is true.